python/chronos/src/bigdl/chronos/data/tsdataset.py

```diff
@@ -812,3 +816,41 @@ def _check_basic_invariants(self, strict_check=False):
         # check dt sorted
         if strict_check:
             _check_dt_is_sorted(self.df, self.dt_col)
+
+    def get_best_lookback(self, aggregate='mode', top_k=3):
+        """
+        Use the calculation of Fourier transform and autocorrelation coefficient to
+        get the time period of all columns, and select the appropriate training time
+        period for the model by specifying aggregate.
+
+        Args:
+            top_k (int): The freq with top top_k power after fft will be
+                used to check the autocorrelation.
+            aggregate (str): Select the mode of calculation time period,
+                We only support 'min', 'max', 'mode', 'median', 'mean'.
+
+        Returns:
+            int: A best time period.
+        """
+        df = self.df.copy()
```

Comment:
why we need a copy here? This will cost a huge amount of mem when the dataset is large